File: server/models/plan.js
const mongoose = require('mongoose');

const PlanSchema = new mongoose.Schema({
  name: {type:String, required:true},
  price: {type:Number, required:true}, // in smallest unit (e.g., paise)
  durationMinutes: {type:Number, required:true}, // e.g., 60 for 1hr
  deviceLimit: {type:Number, default:1},
  visible: {type:Boolean, default:true},
  description: String
}, {timestamps:true});

module.exports = mongoose.model('Plan', PlanSchema);

File: client/lib/models/plan.dart
class Plan {
  final String id;
  final String name;
  final int price;
  final int validity;
  final int data;

  Plan({required this.id, required this.name, required this.price, required this.validity, required this.data});

  factory Plan.fromJson(Map<String, dynamic> json) {
    return Plan(
      id: json['_id'],
      name: json['name'],
      price: json['price'],
      validity: json['validity'],
      data: json['data'],
    );
  }
}
